fuzz/src/chanmon_consistency.rs

```diff
@@ -921,17 +925,29 @@ pub fn do_test<Out: test_logger::Output>(data: &[u8], out: Out) {
 				*monitor_b.update_ret.lock().unwrap() = Ok(());
 				*monitor_c.update_ret.lock().unwrap() = Ok(());
 
-				if let Some((id, _)) = monitor_a.latest_monitors.lock().unwrap().get(&chan_1_funding) {
-					nodes[0].channel_monitor_updated(&chan_1_funding, *id);
+				{
+					let mon_id = monitor_a.latest_monitors.lock().unwrap().get(&chan_1_funding).map(|(id, _)| *id);
+					if let Some(id) = mon_id {
+						nodes[0].channel_monitor_updated(&chan_1_funding, id);
+					}
 				}
-				if let Some((id, _)) = monitor_b.latest_monitors.lock().unwrap().get(&chan_1_funding) {
-					nodes[1].channel_monitor_updated(&chan_1_funding, *id);
+				{
+					let mon_id = monitor_b.latest_monitors.lock().unwrap().get(&chan_1_funding).map(|(id, _)| *id);
+					if let Some(id) = mon_id {
+						nodes[1].channel_monitor_updated(&chan_1_funding, id);
+					}
 				}
```

Comment:
Was the lock held outside of this scope and thus causing a deadlock?

Reply:
Yes, `latest_monitors` is locked again inside `update_channel`, which can be called eventually by the `channel_monitor_updated` call.